feat: filter erc20 transfer list

qiweiii opened the pull request:
related: https://github.com/Magickbase/godwoken_explorer/issues/556

Add filters:
ERC20 transfer list
- filtered by block number
- filtered by from
- filtered by to
- filtered by age

qiweiii commented:
chrome
<img width="544" alt="image" src="https://user-images.githubusercontent.com/32790369/202394456-2b0a8742-e047-4db9-9928-7f8d2b488902.png">

safari
<img width="323" alt="image" src="https://user-images.githubusercontent.com/32790369/202426820-601bea4b-4530-4d2d-b55a-3dac5f42b60c.png">


@Keith-CY, If we use html native date picker, we cannot customize the date picker (the popup calendar) colors, and different browser will have different styles. do you think we should continue to use native input date picker or use another library's datepicker?

Keith-CY commented:
> chrome <img alt="image" width="544" src="https://user-images.githubusercontent.com/32790369/202394456-2b0a8742-e047-4db9-9928-7f8d2b488902.png">
> 
> safari <img alt="image" width="323" src="https://user-images.githubusercontent.com/32790369/202426820-601bea4b-4530-4d2d-b55a-3dac5f42b60c.png">
> 
> @Keith-CY, If we use html native date picker, we cannot customize the date picker (the popup calendar) colors, and different browser will have different styles. do you think we should continue to use native input date picker or use another library's datepicker?



> chrome <img alt="image" width="544" src="https://user-images.githubusercontent.com/32790369/202394456-2b0a8742-e047-4db9-9928-7f8d2b488902.png">
> 
> safari <img alt="image" width="323" src="https://user-images.githubusercontent.com/32790369/202426820-601bea4b-4530-4d2d-b55a-3dac5f42b60c.png">
> 
> @Keith-CY, If we use html native date picker, we cannot customize the date picker (the popup calendar) colors, and different browser will have different styles. do you think we should continue to use native input date picker or use another library's datepicker?

It was supposed to add a custom datetime picker component that has the same functionalities of the native HTML element, not use the native one directly because it's hard to customize.

FrederLu commented:
1、Exactly 30 pieces of data are filtered by block(same problem with filtering by age), but the cursor at the bottom shows that the next page is available (a blank page). If a single page can be displayed in its entirety, make the next page cursor unclickable.
https://godwoken-explorer-qhsc74ar6-magickbase.vercel.app/account/0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6?tab=erc20&before=g3QAAAADZAAMYmxvY2tfbnVtYmVyYgAKtENkAAlsb2dfaW5kZXhhAGQAEHRyYW5zYWN0aW9uX2hhc2h0AAAAA2QACl9fc3RydWN0X19kACJFbGl4aXIuR29kd29rZW5FeHBsb3Jlci5DaGFpbi5IYXNoZAAKYnl0ZV9jb3VudGEgZAAFYnl0ZXNtAAAAIK6uBIQ8zuVx2_jfDA-Z5-BmUY86tC2Mq3XEs_T7EF2d&id=0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6&block_from=701509&block_to=786014
<img width="1244" alt="image" src="https://user-images.githubusercontent.com/25676576/205420963-52598808-d2ca-4c3d-a60b-68ad89374abc.png">

2、When 30 paging is selected in the second paging, the page data does not match the paging requirements (it is still displayed according to 15 paging).
https://godwoken-explorer-qhsc74ar6-magickbase.vercel.app/account/0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6?tab=erc20&block_from=701510&block_to=786014&page_size=30&after=g3QAAAADZAAMYmxvY2tfbnVtYmVyYgAKtL5kAAlsb2dfaW5kZXhhAGQAEHRyYW5zYWN0aW9uX2hhc2h0AAAAA2QACl9fc3RydWN0X19kACJFbGl4aXIuR29kd29rZW5FeHBsb3Jlci5DaGFpbi5IYXNoZAAKYnl0ZV9jb3VudGEgZAAFYnl0ZXNtAAAAIP0VCzKudWG3Cfgw6y3cJOUV1l18tvG99D7-EUHk8Ggn&id=0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6

3、Unqualified data appears when filtering data through a single condition from/to. As a result, the data is also abnormal when combined with other filters.
https://godwoken-explorer-qhsc74ar6-magickbase.vercel.app/account/0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6?tab=erc20&id=0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6&address_from=0x1a9a8e69246896ac1720c1f407309c142c5c3cdc
<img width="1236" alt="image" src="https://user-images.githubusercontent.com/25676576/205427872-4fb7cbfc-5757-44c5-a9d2-26eac451ef67.png">
<img width="1207" alt="image" src="https://user-images.githubusercontent.com/25676576/205427921-4ddaeba5-243e-4e2a-b334-a804d0482142.png">

4、When the data on the page is small, the From is not hidden after filtering the data twice through age.
https://godwoken-explorer-qhsc74ar6-magickbase.vercel.app/account/0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6?tab=erc20&id=0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6&block_from=740324&block_to=785979&age_range_start=2022-11-10T16%3A00%3A00.000Z&age_range_end=2022-11-11T16%3A00%3A00.000Z
<img width="782" alt="image" src="https://user-images.githubusercontent.com/25676576/205428318-2c79936a-8f40-4a62-b7b0-b692cecba786.png">

qiweiii commented:
@FrederLu please check

1. it may be a backend bug, the `metadata.after` should be null in this case. But this can also be fixed in frontend, if selected page size is `>= metadata.total_count`, I will disable the next page button. This problem should happen in all list pages, but fixed it in this PR
2. it's because you are on the second page before changing page size, and changing page size does not go back to page 1, because the `after` and `before` value in url is not being reset after change page size. Also this problem should happen in all list pages, but fixed in this PR
3. fixed by filter the data
4. fixed

FrederLu commented:
1、The page does not display the data from `0x0000000000000000000000000000000000000000`.
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/25676576/206669191-f2cd3472-6ad3-4ee3-8afc-8ff102cf9937.png">
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/25676576/206669718-842b88b9-dd14-49f9-9877-e57164cacbef.png">

2、The filter pop-up window of Age can be the same as block/from/to, and the sliding page will move together after being evoked.
<img width="1310" alt="image" src="https://user-images.githubusercontent.com/25676576/206670302-ec6fb176-2924-4dcf-8c2d-92a933ce867e.png">

qiweiii commented:
Fixed @FrederLu

FrederLu commented:
This problem has now been fixed, but now using the previous data (29 entries) the game markers are not displayed when switching from page 30 to page 15.

>2、When 30 paging is selected in the second paging, the page data does not match the paging requirements (it is still displayed according to 15 paging). https://godwoken-explorer-qhsc74ar6-magickbase.vercel.app/account/0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6?tab=erc20&block_from=701510&block_to=786014&page_size=30&after=g3QAAAADZAAMYmxvY2tfbnVtYmVyYgAKtL5kAAlsb2dfaW5kZXhhAGQAEHRyYW5zYWN0aW9uX2hhc2h0AAAAA2QACl9fc3RydWN0X19kACJFbGl4aXIuR29kd29rZW5FeHBsb3Jlci5DaGFpbi5IYXNoZAAKYnl0ZV9jb3VudGEgZAAFYnl0ZXNtAAAAIP0VCzKudWG3Cfgw6y3cJOUV1l18tvG99D7-EUHk8Ggn&id=0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6
>
<img width="1145" alt="image" src="https://user-images.githubusercontent.com/25676576/206852083-ba36333f-6ac1-4b4e-999c-5bd51cebadc9.png">

Similar questions:
<img width="747" alt="image" src="https://user-images.githubusercontent.com/25676576/206851896-6e7b0889-3d13-4a77-9214-1806a1ee6a6a.png">

qiweiii commented:
Fixed @FrederLu
